deploy/helm/elastic-agent/README.md.gotmpl

```diff
@@ -0,0 +1,58 @@
+<!--
+(NOTE: Do not edit README.md directly. It is a generated file!)
+(      To make changes, please modify README.md.gotmpl and run `helm-docs`)
+-->
+
+{{ template "chart.header" . }}
+{{ template "chart.deprecationWarning" . }}
+
+{{ template "chart.badgesSection" . }}
+
+{{ template "chart.description" . }}
+
+{{ template "chart.homepageLine" . }}
+
+This chart simplifies the deployment of Elastic Agent in Kubernetes and features a built-in Kubernetes policy.
+
+{{ template "chart.maintainersSection" . }}
+
+{{ template "chart.sourcesSection" . }}
+
+{{ template "chart.valuesSection" . }}
+
+{{ define "section.description.Output.PreTable" }}
+The outputs section specifies where to send data. You can specify multiple outputs to pair specific inputs with specific outputs.
+{{ end }}
+
+{{ define "section.description.OutputObject.PreTable" }}
+The supported types of outputs are:
+- `ESPlainAuth`: `elasticsearch` output with the connection details specified inline the yaml
+- `ESSecretAuthBasic`: `elasticsearch` output with the connection details specified in a k8s secret
+- `ESSecretAuthAPI`: `elasticsearch` output with the connection details specified in a k8s secret
+- `ESECKRef`: `elasticsearch` output that references by name an Elasticsearch cluster managed by ECK operator
+{{ end }}
+
+{{ define "section.description.OutputObject.PostTable" }}
+Examples of Helm chart arguments to define an output with name `myOutput`:
+- `ESPlainAuth`: `--set outputs.myOutput.url=https://elasticsearch:9200 --set outputs.myOutput.api_key=token`
+- `ESSecretAuthBasic`: `--set outputs.myOutput.type=ESSecretAuthBasic --set outputs.myOutput.secretName=k8s_secret_name` (required keys in the k8s secret are `url`, `username`, `password`)
+- `ESSecretAuthAPI`: `--set outputs.myOutput.type=ESSecretAuthAPI --set outputs.myOutput.secretName=k8s_secret_name` (required keys in the k8s secret are `url`, `api_key`)
+- `ESECKRef`: `--set outputs.myOutput.type=ESECKRef --set outputs.myOutput.name=eck_es_cluster_name`
```

Comment:
Does this require the cluster to be in the same namespace? Or can it be in a different namespace? If so how can referencing a cluster in another namespace work?

Reply:
I think you are referring to `ESECKRef` and that's actually a good question; If I am not mistaken when you do an reference to an existing cluster the ECK operator can search for it in the namespaces the operator has visibilities, but I am going to test this and come back

Reply:
ty for the proposal @blakerouse, I added a complete example about ECK with a cluster in a different namespace 🙂 feel free to unresolve if you spot anything else